src/agent/src/sandbox.rs

```diff
@@ -151,7 +151,11 @@ impl Sandbox {
     pub fn remove_sandbox_storage(&self, path: &str) -> Result<()> {
         let mounts = vec![path.to_string()];
         remove_mounts(&mounts)?;
-        fs::remove_dir_all(path).context(format!("failed to remove dir {:?}", path))?;
+        // "remove_dir" can fail if it's a read-only filesystem.
```

Comment:
I think we should update the comment now.



```suggestion
        // "remove_dir" will fail if the mount point is backed by a read-only filesystem. This'll be the case with device mapper snapshotter, where we mount the block device directly at the underlying sandbox path which was provided from the base RO kataShared path from the host. 
```
```suggestion
        // "remove_dir" can fail if it's a read-only filesystem.
```